src/MoBi.Presentation/Presenter/ChartPresenter.cs

```diff
@@ -190,54 +207,76 @@ protected virtual void OnDragDrop(object sender, DragEventArgs e)
-            historicalResultsNodes.Each(n => addHistoricalResults(n.Tag));
+            addHistoricalResults(historicalResultsNodes.Select(result => result.Tag).ToList());
          }
          else
          {
             var droppedObservedData = _observedDataDragDropBinder.DroppedObservedDataFrom(e);
-            droppedObservedData.Each(addObservedData);
+            addObservedData(droppedObservedData.ToList());
          }
       }
 
       protected abstract bool CanDropSimulation { get; }
 
       protected abstract void MarkChartOwnerAsChanged();
 
-      private void addHistoricalResults(DataRepository repository)
+      private void addHistoricalResults(IReadOnlyList<DataRepository> repositories)
       {
-         addDataRepositorySimulation(repository);
-         _editorPresenter.AddDataRepository(repository);
-         repository.SetPersistable(persistable: true);
+         addDataRepositoriesToSimulation(repositories);
+         _editorPresenter.AddDataRepositories(repositories);
+         repositories.Each(repository => repository.SetPersistable(persistable: true));
       }
 
-      private void addObservedData(DataRepository repository)
+      private void addObservedData(IReadOnlyList<DataRepository> repositories)
       {
-         _editorPresenter.AddDataRepository(repository);
-         repository.Where(c => c.DataInfo.Origin == ColumnOrigins.Observation)
-            .Each(c => _editorPresenter.AddCurveForColumn(c.Id));
+         _editorPresenter.AddDataRepositories(repositories);
+         using (_chartUpdater.UpdateTransaction(Chart))
+         {
+            repositories.SelectMany(x => x.ObservationColumns()).Each(observationColumn => _editorPresenter.AddCurveForColumn(observationColumn));
+         }
       }
 
       private void addObservedDataIfNeeded(IEnumerable<DataRepository> dataRepositories)
       {
          //curves are already selected. no need to add default selection
          if (Chart.Curves.Any()) return;
-         dataRepositories.Each(addObservedData);
+         addObservedData(dataRepositories.ToList());
       }
 
-      public ICurveChart Chart
+      public CurveChart Chart
       {
-         get { return _displayPresenter.DataSource; }
-         protected set
-         {
-            if (Chart != null)
-               Chart.PropertyChanged -= onPropertyChanged;
+         get => _displayPresenter.Chart;
+         protected set => Edit(value);
+      }
 
-            _editorPresenter.DataSource = value;
-            _displayPresenter.DataSource = value;
+      public void Clear()
+      {
+         releasePropertyChanged();
 
-            if (Chart != null)
-               Chart.PropertyChanged += onPropertyChanged;
-         }
+         _editorPresenter.Clear();
+         _displayPresenter.Clear();
+      }
+
+      private void releasePropertyChanged()
+      {
+         if (Chart != null)
+            Chart.PropertyChanged -= onPropertyChanged;
```

Comment:
remove that. That should not be used anymore and won't be fired as it used to anyways